File: NumberTranslatorStaticClassesApiTests/NumberTests.cs
using System;
using NumberTranslatorStaticClassesApi.Enums;
using NUnit.Framework;

using NumberTranslatorStaticClassesApi.Services;

namespace NumberTranslatorStaticClassesApiTests
{
    public class Tests
    {
        [SetUp]
        public void Setup()
        {
        }

        [Test]
        public void TestGetDigit()
        {
            var expected = new[]{ 3, 1 };
            var actual = new[]{ DigitService.GetDigit(DigitPlacesEnum.Tens, 31),
                                     DigitService.GetDigit(DigitPlacesEnum.Ones, 31)};

            Assert.AreEqual(expected, actual);
        }

        [Test]
        public void TestUniqueNumbers()
        {
            static Func<int, string> TranslateUniqueNumbersWithRepoMock()
            {
                return x => UniqueNumberTranslatorService.TranslateUniqueNumbers(_ => "zero", x);
            }

            var expected = "zero";
            var actual = TranslateUniqueNumbersWithRepoMock()(0);

            Assert.AreEqual(expected, actual);
        }

        [Test]
        public void TestTeens()
        {
            var expected = "thirteen";
            string GetPrefixMock(int _) => "thir";
            var actual = TeensTranslatorService.TranslateTeens(GetPrefixMock, 13);

            Assert.AreEqual(expected, actual);
        }

        [Test]
        public void TestTranslateTensWithZeroAtEnd()
        {
            var expected = "twenty";

            static Func<int, string> TranslateUniqueNumbers()
            {
                return x => "";
            }

            static Func<int, string> GetPrefixService()
            {
                return x => "twen";
            }

            var actual = TensTranslatorService.TranslateTens(GetPrefixService(), TranslateUniqueNumbers(), 20);

            Assert.AreEqual(expected, actual);
        }

        [Test]
        public void TestTranslateTensWithNonZeroAtEnd()
        {
            var expected = "twenty-two";

            static Func<int, string> TranslateUniqueNumbers()
            {
                return x => "two";
            }

            static Func<int, string> GetPrefixService()
            {
                return x => "twen";
            }

            var actual = TensTranslatorService.TranslateTens(GetPrefixService(), TranslateUniqueNumbers(), 22);

            Assert.AreEqual(expected, actual);
        }
    }
}

File: FSharpNumberTranslatorApiTests/NumberTests.fs
module FSharpNumberTranslatorApiTests

open NUnit.Framework
open IntExtensions
open UniqueNumberTranslatorService
open TeensTranslatorService
open TensTranslatorService

[<Test>]
let TestGetDigit() =
    let expected = [|3; 1|]
    let actual = [|(31 |> GetDigit DigitPlaces.Tens);
                   (31 |> GetDigit DigitPlaces.Ones)|]

    Assert.AreEqual(expected, actual)

[<Test>]
let TestUniqueNumbers() =    
    let TranslateUniqueNumbersWithRepo =
        TranslateUniqueNumbers (fun _ -> "zero")

    let expected = "zero"
    let actual = 0 |> TranslateUniqueNumbersWithRepo
    
    Assert.AreEqual(expected, actual)

[<Test>]
let TestTeens() =
    let expected = "thirteen"
    let actual = 13 |> TranslateTeens (fun _ -> "thir")

    Assert.AreEqual(expected, actual)

[<Test>]
let TestTranslateTensWithZeroAtEnd() =
    let expected = "twenty"
    let translateUniqueNumbers = (fun _ -> "")
    let getPrefixService = (fun _ -> "twen")
    let actual = 20 |> TranslateTens getPrefixService translateUniqueNumbers

    Assert.AreEqual(expected, actual)

[<Test>]
let TestTranslateTensWithNonZeroAtEnd() =
    let expected = "twenty-two"
    let translateUniqueNumbers = (fun _ -> "two")
    let getPrefix = (fun _ -> "twen")
    let actual = 22 |> TranslateTens getPrefix translateUniqueNumbers

    Assert.AreEqual(expected, actual)